lib/prima_auth0_ex/token_provider/cached_token_service.ex

```diff
@@ -9,28 +9,26 @@ defmodule PrimaAuth0Ex.TokenProvider.CachedTokenService do
   one from the authorization provider.
   """
   alias PrimaAuth0Ex.Config
+  alias PrimaAuth0Ex.TokenCache
 
   alias PrimaAuth0Ex.TokenProvider.{
     Auth0AuthorizationService,
-    EncryptedRedisTokenCache,
     TokenService
   }
 
-  alias PrimaAuth0Ex.TokenCache.EncryptedRedisTokenCache
-
   @behaviour TokenService
 
   @impl TokenService
   def retrieve_token(credentials, audience) do
     credentials.client
-    |> token_cache().get_token_for(audience)
+    |> TokenCache.get_token_for(audience)
```

Comment:
Can a behavior be used directly like this?

Reply:
if you look at the implementation of TokenCache it's a module that defines a behavior but also implements the behavior itself, delegating the actual implementation of the functions to whatever is set in the config

Reply:
I'm dumb